pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1289,39 +1290,13 @@ func doActivate(ctx *domainContext, config types.DomainConfig,
 	}
 	defer file.Close()
 
-	if err := configToXencfg(config, *status, ctx.assignableAdapters,
-		file); err != nil {
+	if err := hyper.CreateDomConfig(status.DomainName, config, status.DiskStatusList, ctx.assignableAdapters, file); err != nil {
```

Comment:
Probably need a second step of refactoring since the existence of the xen config file and its path name is on line 1286. The existence of such a file might need to be hidden under the hyper abstraction.